src/lib/pickles/composition_types/composition_types.mli

```diff
@@ -0,0 +1,1708 @@
+    that the step-circuit partially verifies. *)
+      ; messages_for_next_step_proof : 'messages_for_next_step_proof
+            (** The component of the proof accumulation state that is only computed on by the
+          "stepping" proof system, and that can be handled opaquely by any "wrap" circuits. *)
+      }
+    [@@deriving sexp, compare, yojson, hlist]
+
+    val spec :
+         ('a, 'b, 'c) Spec.T.t
+      -> ('d, 'e, 'c) Spec.T.t
+      -> ( ('a * ('d * unit)) Pickles_types.Hlist.HlistId.t
+         , ('b * ('e * unit)) Pickles_types.Hlist.HlistId.t
+         , 'c )
+         Spec.T.t
+
+    val to_data :
+         ( ( ( 'a
+             , 'b
+             , 'c
+             , 'd
+             , 'e Pickles_types.Hlist0.Id.t
+             , 'f
+             , 'g )
+             Per_proof.In_circuit.t
+           , 'h )
+           Pickles_types.Vector.t
+         , 'i Pickles_types.Hlist0.Id.t )
+         t
+      -> ( (    option_map:
+                  (   'd
+                   -> f:
+                        (   ( 'j Pickles_types.Hlist0.Id.t
+                            , 'k Pickles_types.Hlist0.Id.t )
+                            Deferred_values.Plonk.In_circuit.Lookup.t
+                         -> ('j * ('k * unit)) Pickles_types.Hlist.HlistId.t )
+                   -> 'l Pickles_types.Hlist0.Id.t )
+             -> ( ( 'c
+                  , Pickles_types.Nat.N9.n Pickles_types.Nat.s
+                    Pickles_types.Nat.s
+                    Pickles_types.Nat.s
+                    Pickles_types.Nat.s
+                    Pickles_types.Nat.s
+                    Pickles_types.Nat.s
+                    Pickles_types.Nat.s
+                    Pickles_types.Nat.s
+                    Pickles_types.Nat.s
+                    Pickles_types.Nat.s )
+                  Pickles_types.Vector.t
+                * ( ( 'f
+                    , Pickles_types.Nat.z Pickles_types.Nat.s )
+                    Pickles_types.Vector.t
+                  * ( ( 'a
+                      , Pickles_types.Nat.z Pickles_types.Nat.s
+                        Pickles_types.Nat.s )
+                      Pickles_types.Vector.t
+                    * ( ( 'b
+                        , Pickles_types.Nat.z Pickles_types.Nat.s
+                          Pickles_types.Nat.s
+                          Pickles_types.Nat.s )
+                        Pickles_types.Vector.t
+                      * ( 'e
+                        * ( ( 'g
+                            , Pickles_types.Nat.z Pickles_types.Nat.s )
+                            Pickles_types.Vector.t
+                          * ('l * unit) ) ) ) ) ) )
+                Pickles_types.Hlist.HlistId.t
+           , 'h )
+           Pickles_types.Vector.t
+         * ('i * unit) )
+         Pickles_types.Hlist.HlistId.t
+
+    val of_data :
+         ( ( ( ( 'a
+               , Pickles_types.Nat.N9.n Pickles_types.Nat.s Pickles_types.Nat.s
```

Comment:
Can you replace this (and other repeated `Nat.s`s) with the appropriate number (e.g. `Nat.N19.n` here)?

Reply:
Done